File: src/csharp/api/Repositories/IOrderRepository.cs
using Example.Api.Models;

namespace Example.Api.Repositories;

public interface IOrderRepository
{
    /// <summary>
    /// Get patient's order by id.
    /// </summary>
    /// <param name="id">The order id.</param>
    /// <returns>The order entity.</returns>
    Task<PatientOrder?> GetOrderAsync(long id);

    /// <summary>
    /// Creates a new patient's order.
    /// </summary>
    /// <param name="order">The order entity to create.</param>
    /// <returns>The created order entity.</returns>
    Task<PatientOrder> AddAsync(PatientOrder order);

    /// <summary>
    /// Update the message of an existing order by id.
    /// </summary>
    /// <param name="order">The order entity with updated message.</param>
    /// <returns>The updated order entity.</returns>
    Task<PatientOrder> UpdateMessageAsync(PatientOrder order);

    /// <summary>
    /// Update the message of an order.
    /// </summary>
    /// <param name="id">The id of the order to update.</param>
    /// <param name="message">The new message for the order.</param>
    /// <param name="updatedAt">The timestamp of the update.</param>
    /// <returns>The updated order entity, or null if not found.</returns>
    Task<PatientOrder?> UpdateAsync(long id, string message, DateTimeOffset updatedAt);
}

File: src/nodejs/src/repositories/IOrderRepository.ts
import { Order } from "../entities/Order";

/**
 * Interface for order repository operations.
 */
export interface IOrderRepository {
  /**
   * Retrieves an order by its ID.
   * @param id - The order ID.
   * @returns The order if found, null otherwise.
   */
  getOrderAsync(id: number): Promise<Order | null>;

  /**
   * Adds a new order to the repository.
   * @param order - The order to add.
   * @returns The added order with generated ID.
   */
  addAsync(order: Order): Promise<Order>;

  /**
   * Updates an order's message.
   * @param id - The order ID.
   * @param message - The new message.
   * @param updatedAt - The update timestamp.
   * @returns The updated order if found, null otherwise.
   */
  updateAsync(id: number, message: string, updatedAt: Date): Promise<Order | null>;
}
